subprojects/logging/src/main/java/org/gradle/util/SingleMessageLogger.java

```diff
@@ -59,102 +59,105 @@ public synchronized static void reportSuppressedDeprecations() {
+            nagUserWith(String.format("The %s method has been deprecated.", methodName), thisWillBeRemovedMessage(), String.format("Please use the %s method instead.", replacement));
         }
     }
 
     public static void nagUserOfReplacedProperty(String propertyName, String replacement) {
         if (isEnabled()) {
             nagUserWith(String.format(
-                    "The %s property %s. Please use the %s property instead.",
-                    propertyName, getDeprecationMessage(), replacement));
+                    "The %s property has been deprecated.", propertyName), thisWillBeRemovedMessage(), String.format("Please use the %s property instead.", replacement));
         }
     }
 
     public static void nagUserOfDiscontinuedMethod(String methodName) {
         if (isEnabled()) {
-            nagUserWith(String.format("The %s method %s.",
-                    methodName, getDeprecationMessage()));
+            nagUserWith(String.format("The %s method has been deprecated.", methodName), thisWillBeRemovedMessage(), null);
         }
     }
 
     public static void nagUserOfDiscontinuedMethod(String methodName, String advice) {
         if (isEnabled()) {
-            nagUserWith(String.format("The %s method %s. %s",
-                    methodName, getDeprecationMessage(), advice));
+            nagUserWith(String.format("The %s method has been deprecated.", methodName), thisWillBeRemovedMessage(), advice);
         }
     }
 
     public static void nagUserOfDiscontinuedProperty(String propertyName, String advice) {
         if (isEnabled()) {
-            nagUserWith(String.format("The %s property %s. %s",
-                    propertyName, getDeprecationMessage(), advice));
+            nagUserWith(String.format("The %s property has been deprecated.", propertyName), thisWillBeRemovedMessage(), advice);
         }
     }
 
     public static void nagUserOfDiscontinuedApi(String api, String advice) {
         if (isEnabled()) {
-            nagUserWith(String.format("The %s %s. %s",
-                api, getDeprecationMessage(), advice));
+            nagUserWith(String.format("The %s has been deprecated.", api), thisWillBeRemovedMessage(), advice);
         }
     }
 
     public static void nagUserOfReplacedNamedParameter(String parameterName, String replacement) {
         if (isEnabled()) {
-            nagUserWith(String.format(
-                    "The %s named parameter %s. Please use the %s named parameter instead.",
-                    parameterName, getDeprecationMessage(), replacement));
+            nagUserWith(String.format("The %s named parameter has been deprecated.", parameterName), thisWillBeRemovedMessage(), String.format("Please use the %s named parameter instead.", replacement));
+        }
+    }
+
+    /**
+     * Try to avoid using this nagging method. The other methods use a consistent wording for when things will be removed.
+     */
+    static void nagUserWith(String message) {
```

Comment:
Why is this package-private? If this method is removed on purpose, why not private? However, my personal preference would be keeping it public like other overloads in case some plugin authors are using it - they usually have no idea `org.gradle.util` is not a public package.

Reply:
We should not keep it because users might use it. This class is very much not public, and we need to be able to change it.

I like the user empathy, but in this case would should not speculatively keep it.

Reply:
I kept it package private as its used only in some tests atm. I'll add a comment about this

Reply:
You can use `@VisibleForTesting` to express that